aptos-move/framework/aptos-framework/sources/stake.spec.move

```diff
@@ -0,0 +1,73 @@
+spec aptos_framework::stake {
+    spec calculate_rewards_amount {
+        pragma opaque;
+        ensures [concrete] (rewards_rate_denominator * num_total_proposals == 0) ==> result == 0;
+        ensures [concrete] (rewards_rate_denominator * num_total_proposals > 0) ==>
+                           result == ((stake_amount * rewards_rate * num_successful_proposals) /
```

Comment:
I wonder if we just make assumptions that stake_amount < u64.max, rewards_rate < 100 and num_successful_proposals < 86400 (1 proposal per second in a day) and just have an ensure that the reward calc doesn't overflow?

Reply:
Could we add assertions for "rewards_rate < 100 and num_successful_proposals < 86400"? At least, I will add this rationale as a comment.

Reply:
Feel free to add assertion for rewards rate in https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/configs/staking_config.move#L139. num_successful_proposal might be a bit trickier as it depends on the caller of block#block_prologue or effectively the rate of block production on the network.

Reply:
@movekevin , I added an assertion into the function `update_rewards_rate`.